cmd/ncproxy/ncproxy.go

```diff
@@ -557,6 +558,58 @@ func (s *grpcService) GetNetworks(ctx context.Context, req *ncproxygrpc.GetNetwo
 // client connections.
 type ttrpcService struct {
 	m sync.Mutex
+	// database for containerID to compute agent address
+	agentStore *computeAgentStore
+}
+
+func newTTRPCService(ctx context.Context, db *bolt.DB) *ttrpcService {
+	agentStore := newComputeAgentStore(db)
+	return &ttrpcService{
+		agentStore: agentStore,
+	}
+}
+
+func getComputeAgentClient(agentAddr string) (computeagent.ComputeAgentService, error) {
```

Comment:
If we have many compute agents that need to be reconnected to (alive or dead), it seems this function could end up taking a while. Do we need to do anything to make sure other components are aware that ncproxy is still initializing?

Reply:
how so? As in make sure that other components adjust their timeouts (if any) to account for this? I could also have this called in parallel below.

Reply:
Maybe a good question would be:

> If ncproxy is down for a while (e.g. stuck attempting to reconnect to dead containers), what happens to other components on the system that are trying to contact it? Do they have some way of knowing when ncproxy is up, do they need timeouts adjusted, etc.

Reply:
I've updated the code to attempt to reconnect to the compute agents in parallel. I'm not sure if there is something else to help make sure ncproxy is still starting up. If anyone has suggestions, please let me know.